azure/Kqlmagic/draft_client.py

```diff
@@ -41,12 +41,30 @@ class DraftClient(object):
     _API_VERSION = "v1"
     _GET_SCHEMA_QUERY = ".show schema"
 
-    def __init__(self, conn_kv: dict, domain: str, data_source: str):
+    _CLOUD_AAD_URLS={
+    "public": "https://api.{0}.io",
+    "mooncake":"https://api.{0}.azure.cn",
+    "fairfax":"https://api.{0}.us",
+    "blackforest":"https://api.{0}.de",
+}
+
+
+
+    def __init__(self, conn_kv: dict, domain: str, data_source: str, **options):
+
+        cloud = options.get("cloud") or "public"
         self._domain = domain
-        self._data_source = data_source
+        if data_source.find("loganalytics") >= 0:
+            self._data_source = self._CLOUD_AAD_URLS.get(cloud).format("loganalytics")
+        elif data_source.find("applicationinsights") >= 0:
+            self._data_source = self._CLOUD_AAD_URLS.get(cloud).format("applicationinsights")            
```

Comment:
the engine should pass the original public url, see the comment above on _CLOUD_AAD_URLS

Reply:
so there is no need to update data source in draft client?